pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqluser/_generated_object_sqluser.golden.yaml

```diff
@@ -26,7 +26,6 @@ spec:
         key: password
         name: sqluser-dep-${uniqueId}
   resourceID: sqluser-${uniqueId}
-  type: BUILT_IN
```

Comment:
I guess(?) this is a default optional field. Can we keep the server's behavior here?

Reply:
It looks like GCP does not return this field by default. This was probably originally added due to `state-into-spec: merge`, combined with the mock http mux marshaler previously being configured to return "0" / nil / empty values (this is the "0" value for the enum). It should be removed, since that's the GCP behavior.

Reply:
Does this git diff come from the real GCP and MockGCP (that the previous object and http run against real GCP and this PR switched to mockgcp )? If so, the `state-into-spec` will not make a diff, we can tell that from the  _http.log response [here](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2324/files#diff-187f6b6f9b12d8c78beee54344ae65366049302eb4f6b2bd984448cb04bc3f90L91). I'm okay with some fields not handled in one PR, especially since this PR already contains a lot of good stuff and is really big now. Just want to make sure I'm on the same page about where the changes come from.

Reply:
The reason for this diff is because the sqluser _http.log and golden.yaml were based on mockgcp. Real GCP does not return `type: BUILT_IN`. The only reason mockgcp was returning BUILT_IN was because it's a 0 value, and before this change, the mockgcp sql service (unlike real gcp) was configured to return all 0 values. Here's where the code defines BUILT_IN as the 0 value for that field: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/e29dd609e93ff648b0f6e06a45317d6b2e1b0cf2/mockgcp/generated/mockgcp/cloud/sql/v1beta4/cloud_sql_users.pb.go#L45. So, in this case, I think it's correct to remove this, so we can match the real GCP behavior.